Client/Frontend/Settings/HomepageSettings/HomePageSettingViewController.swift

```diff
@@ -280,10 +280,17 @@ extension HomePageSettingViewController {
         }
 
         override func onClick(_ navigationController: UINavigationController?) {
-            let viewModel = LegacyWallpaperSettingsViewModel(with: tabManager,
-                                                       and: LegacyWallpaperManager())
-            let wallpaperVC = LegacyWallpaperSettingsViewController(with: viewModel)
-            navigationController?.pushViewController(wallpaperVC, animated: true)
+            if let wallpaperVersion: WallpaperVersion = featureFlags.getCustomState(for: .wallpaperVersion),
+               wallpaperVersion == .v2 {
+                let viewModel = WallpaperSettingsViewModel(wallpaperManager: WallpaperManager(), tabManager: tabManager)
+                let wallpaperVC = WallpaperSettingsViewController(viewModel: viewModel)
+                navigationController?.pushViewController(wallpaperVC, animated: true)
+            } else {
+                let viewModel = LegacyWallpaperSettingsViewModel(with: tabManager,
+                                                           and: LegacyWallpaperManager())
+                let wallpaperVC = LegacyWallpaperSettingsViewController(with: viewModel)
+                navigationController?.pushViewController(wallpaperVC, animated: true)
+            }
```

Comment:
This is the third time I'm seeing this code repeated. Perhaps a minor improvement for maintainability would be to have a method somewhere that returns a VC, and then just push the VC with whatever is pushing/presenting the view?